File: 0x08-python-more_classes/4-rectangle.py
#!/usr/bin/python3
"""
Define a Rectangle Class
"""


class Rectangle:
    def __init__(self, width=0, height=0):
        self.__width = width
        self.__height = height

    @property
    def width(self):
        return self.__width

    @width.setter
    def width(self, value):
        if not isinstance(value, int):
            raise TypeError("width must be an integer")
        if value < 0:
            raise ValueError("width must be >=0")
        self.__width = value

    @property
    def height(self):
        return self.__height

    @height.setter
    def height(self, value):
        if not isinstance(value, int):
            raise TypeError("height must be an integer")
        if value < 0:
            raise ValueError("height must be >=0")
        self.__height = value

    def area(self):
        return (self.__width * self.__height)

    def perimeter(self):
        if self.__width == 0 or self.__height == 0:
            return(0)
        else:
            return (self.__width + self.__width + self.height + self.height)

    def __str__(self):
        if self.__width == 0 or self.__height == 0:
            return("")
        else:
            rectangle = "#"
            for i in range(self.__height):
                for j in range(self.__height - 1):
                    print(rectangle * self.__width)
                return(rectangle * self.width)

    def __repr__(self):
        return("Rectangle({}, {})".format(self.width, self.height))

File: 0x13-javascript_objects_scopes_closures/4-rectangle.js
#!/usr/bin/node

exports.Rectangle = function Rectangle (w, h) {
  if ((w !== undefined && w > 0) && (h !== undefined && h > 0)) {
    this.width = w;
    this.height = h;
  }
  this.print = function () {
    for (let x = 0; x < this.height; x++) {
      console.log('X'.repeat(this.width));
    }
  };

  this.rotate = function () {
    let temp = this.width;
    this.width = this.height;
    this.height = temp;
  };

  this.double = function () {
    this.width *= 2;
    this.height *= 2;
  };
};
